### PR TITLE
Add module container test

### PR DESCRIPTION
**Pull request by yiiliveext**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: tests/Unit/AbstractModuleContainerTest.php [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+use Yiisoft\Di\Tests\Support\Car;
+use Yiisoft\Di\Tests\Support\CarInterface;
+use Yiisoft\Di\Tests\Support\EngineInterface;
+use Yiisoft\Di\Tests\Support\EngineMarkOne;
+use Yiisoft\Di\Tests\Support\EngineMarkTwo;
+use Yiisoft\Factory\Definitions\Reference;
+
+abstract class AbstractModuleContainerTest extends TestCase
+{
+    abstract public function createModuleContainer(array $rootDefinitions, array $moduleDefinitions): array;
```
Comment (samdark) [inline tests/Unit/AbstractModuleContainerTest.php:17]: Are there going to be multiple implementations?

File: tests/Unit/AbstractModuleContainerTest.php [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+use PHPUnit\Framework\TestCase;
+use Yiisoft\Di\Tests\Support\Car;
+use Yiisoft\Di\Tests\Support\CarInterface;
+use Yiisoft\Di\Tests\Support\EngineInterface;
+use Yiisoft\Di\Tests\Support\EngineMarkOne;
+use Yiisoft\Di\Tests\Support\EngineMarkTwo;
+use Yiisoft\Factory\Definitions\Reference;
+
+abstract class AbstractModuleContainerTest extends TestCase
```
Comment (samdark) [inline tests/Unit/AbstractModuleContainerTest.php:15]: Tests in this class are correct but it looks like some are duplicating each other.

File: tests/Unit/ModuleContainerTest.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+use Psr\Container\ContainerInterface;
+use Yiisoft\Di\Container;
+use Yiisoft\Di\CompositeContainer;
+
+class ModuleContainerTest extends AbstractModuleContainerTest
+{
+    public function createModuleContainer(array $rootDefinitions, array $moduleDefinitions): array
+    {
+        $compositeContainer = new CompositeContainer();
+        //$rootContainer = new Container($rootDefinitions, [],  $compositeContainer);
```
Comment (samdark) [inline tests/Unit/ModuleContainerTest.php:16]: Remove?

Comment (yiiliveext) [reply tests/Unit/ModuleContainerTest.php:16]: This is just an example of a module container. We haven't any valid implementation now, so I wrote the abstract test.

File: tests/Unit/AbstractModuleContainerTest.php (hunk 2 above)
Comment (yiiliveext) [reply tests/Unit/AbstractModuleContainerTest.php:15]: They have many small but important differences.

**Comment by samdark** [comment]
This PR will be updated soon. Problems in tests are valid and to be solved.

**Pull request closed without merging**